### PR TITLE
Add facility reviews

### PR DESCRIPTION
**Pull request by mathias234**
Add the FacilityReview model
This model holds a review for a specific facility in a space.

This PR also contains a first stab at implementing a system for aggregating facility reviews.

### CODE REVIEW

File: app/services/spaces/aggregate_facility_reviews.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    def initialize(space:)
+      @space = space
+      super()
+    end
+
+    def call # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+      space.aggregated_facility_reviews.destroy_all
+      space.reload
+
+      a = space.facilities.order(:created_at).map do |facility|
```
[the remark this answers is not in the document]
Comment (DanielJackson-Oslo) [reply app/services/spaces/aggregate_facility_reviews.rb:14]: (Same goes for agg and a later on line 22+ )

**Review by DanielJackson-Oslo — COMMENTED** [review]
Cool work! (See comment about overly shortened variable names. To me they make those parts of the code less readable than it ought to be)

I think we can do aggregation a bit more nuanced, either in it's own PR or before this one merges. How we would aggregate wasn't specified, and I like this first stab at it for it's simplicity!

## Aggregation as % of positive reviews 

In my mind it was handled something like this, but that might add more complexity than it's worth?

We take the last 10 reviews. 

Out of those, how many were positive?

If the ratio is above 80%, set it as likely. 

If the ratio is below 30% set it as unlikely.

Anything else, set it as maybe. 

If no reviews, it is unknown (obv.)

## They don't have this at all

Then there's the special case of "They don't have this at all", or `was_not_available` in your code. 

That's not merely a negative result, but more like a "you won't ever be allowed, as this facility does not even exist". 

It should probably have it's own aggregation state beside unknown, unlikely, maybe and likely. Something like `not existing`, and if enough people say it doesn't exist (over 50%?), then it should be marked as `not existing` - not only `not allowed to use`.

Does that make sense?

**Comment by mathias234** [comment]
> Cool work! (See comment about overly shortened variable names. To me they make those parts of the code less readable than it ought to be)
> 
> I think we can do aggregation a bit more nuanced, either in it's own PR or before this one merges. How we would aggregate wasn't specified, and I like this first stab at it for it's simplicity - but think it might be better to aggregate based on a % of positive reviews, and not just on the binary existence of a single negative review.
> 
> ## Aggregation as % of positive reviews
> In my mind it was handled something like this, but that might add more complexity than it's worth?
> 
> We take the last 10 reviews.
> 
> Out of those, how many were positive?
> 
> If the ratio is above 80%, set it as likely.
> 
> If the ratio is below 30% set it as unlikely.
> 
> Anything else, set it as maybe.
> 
> If no reviews, it is unknown (obv.)
> 
> ## They don't have this at all
> Then there's the special case of "They don't have this at all", or `was_not_available` in your code.
> 
> That's not merely a negative result, but more like a "you won't ever be allowed, as this facility does not even exist".
> 
> It should probably have it's own aggregation state beside unknown, unlikely, maybe and likely. Something like `impossible`. I've added code to reflect that in my PR #3 . If enough people say it doesn't exist (over 50%?), then it should be marked as `impossible` - not only `not allowed to use`.
> 
> Does that make sense?

**RE: Aggregation as % of positive reviews**
I do think this makes sense, but it is probably an improvement we can do in the future, maybe once we have some proper data to test with and tweak the percentages. 

**RE: They don't have this at all**
I think it makes sense to have an `impossible` state, I will add that to the model and the service

File: app/services/spaces/aggregate_facility_reviews.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathias234) [reply app/services/spaces/aggregate_facility_reviews.rb:14]: Yeah probably makes sense to write the full name :)

**Comment by DanielJackson-Oslo** [comment]
> **RE: Aggregation as % of positive reviews**
> I do think this makes sense, but it is probably an improvement we can do in the future, maybe once we have some proper data to test with and tweak the percentages.
> 
> **RE: They don't have this at all**
> I think it makes sense to have an `impossible` state, I will add that to the model and the service

Sounds like a good solution! 👍

**Comment by asolvang** [comment]
[Add facility reviews](https://trello.com/c/PF4H1rAz/25-add-facility-reviews)

**Review by DanielJackson-Oslo — APPROVED** [review]
Looks good to me now!

**Pull request merged**